pallets/ethereum-transactions/src/tests/tests_eth_transaction_type.rs

```diff
@@ -28,21 +33,24 @@ fn generate_publish_root_eth_txn_desc(root_hash: [u8; 32]) -> EthTransactionDesc
         call_values: vec![Token::FixedBytes(root_hash.to_vec())],
     }
 }
-
 fn generate_deregister_validator_eth_txn_desc(
+    t1_public_key: H512,
     t2_public_key: [u8; 32],
 ) -> EthTransactionDescription {
     EthTransactionDescription {
         function_call: Function {
             name: String::from("deregisterValidator"),
-            inputs: vec![Param {
-                name: String::from("_targetT2PublicKey"),
-                kind: ParamType::FixedBytes(32),
-            }],
+            inputs: vec![
+                Param { name: String::from("_targetT2PublicKey"), kind: ParamType::FixedBytes(32) },
+                Param { name: String::from("_targetT1PublicKey"), kind: ParamType::FixedBytes(64) },
+            ],
```

Comment:
I think this is wrong. The call in the contract is this:
`  function deregisterValidator(bytes memory t1PublicKey, bytes32 t2PublicKey, uint256 t2TransactionId,
      bytes calldata confirmations)`
(https://github.com/Aventus-Network-Services/avn-bridge/blob/master/contracts/AVNBridge.sol#L345)

So, the first argument is the T1 public key, and the second is the T2 public key. These arguments are reversed in here.

True, the hash is computed the other way around, but that does not have an impact in the call.